core/server/src/main/java/alluxio/web/UIWebServer.java

```diff
@@ -65,29 +69,36 @@ public UIWebServer(ServiceType service, InetSocketAddress address) {
     QueuedThreadPool threadPool = new QueuedThreadPool();
     int webThreadCount = Configuration.getInt(PropertyKey.WEB_THREADS);
 
-    mServer = new Server();
-    SelectChannelConnector connector = new SelectChannelConnector();
-    connector.setHost(address.getHostName());
-    connector.setPort(address.getPort());
-    connector.setAcceptors(webThreadCount);
-    mServer.setConnectors(new Connector[] {connector});
-
-    try {
-      mServer.getConnectors()[0].close();
-      mServer.getConnectors()[0].open();
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
-    }
-
     // Jetty needs at least (1 + selectors + acceptors) threads.
     threadPool.setMinThreads(webThreadCount * 2 + 1);
     threadPool.setMaxThreads(webThreadCount * 2 + 100);
-    mServer.setThreadPool(threadPool);
+
+    mServer = new Server(threadPool);
+
+    mServerConnector = new ServerConnector(mServer);
+    mServerConnector.setPort(mAddress.getPort());
+    mServerConnector.setHost(mAddress.getAddress().getHostAddress());
+
+    mServer.addConnector(mServerConnector);
```

Comment:
if we call `mServerConnector.open()` here, could we continue to reset the master and worker web ports in their respective constructors? Ideally we could have symmetry between how we create and start the thrift, web, and data servers.